File: February 2024/maximum_sum_problem.py
class Solution:
    def maxSum(self, n, memo={}): 
        if n == 0 or n == 1:
            return n
        if n in memo:
            return memo[n]
        memo[n] = max(n, self.maxSum(n // 2, memo) + self.maxSum(n // 3, memo) + self.maxSum(n // 4, memo))
        
        return memo[n]

File: February 2024/Maximum Sum Problem.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution
{
    public:
      unordered_map<int, int> memo;

    int maxSum(int n) {
     
        if (n == 0) return 0;
        if (memo.find(n) != memo.end()) return memo[n];
        int sum = maxSum(n / 2) + maxSum(n / 3) + maxSum(n / 4);
        memo[n] = max(n, sum);
        return memo[n];
    }
};

//{ Driver Code Starts.
int main()
{
    int t;
    cin>>t;
    while(t--)
    {
        int n;
        cin>>n;
        Solution ob;
        cout<<ob.maxSum(n)<<"\n";
    }
    return 0;
}
// } Driver Code Ends
